services/src/main/java/org/apache/unomi/services/impl/rules/RulesServiceImpl.java

```diff
@@ -277,10 +278,23 @@ public void refreshRules() {
 
     private List<Rule> getAllRules() {
         List<Rule> rules = persistenceService.getAllItems(Rule.class, 0, -1, "priority").getList();
+        List<Rule> rulesToSwitch = new ArrayList<>();
         for (Rule rule : rules) {
-            ParserHelper.resolveConditionType(definitionsService, rule.getCondition(), "rule " + rule.getItemId());
-            ParserHelper.resolveActionTypes(definitionsService, rule);
+            // Check rule integrity
+            boolean isValid = ParserHelper.resolveConditionType(definitionsService, rule.getCondition(), "rule " + rule.getItemId());
+            isValid = isValid && ParserHelper.resolveActionTypes(definitionsService, rule);
+            // check if rule status has changed
+            if (rule.getMetadata().isEnabled() != isValid) {
+                rule.getMetadata().setEnabled(isValid);
+                rulesToSwitch.add(rule);
+            }
```

Comment:
Something is wrong here, in case:
- rule enabled: false
- rule valid: true

Then it will force enable the rule and save it as enabled, soometimes users just want to create draft rule that could be valid but not enabled.